Kernel/Arch/x86_64/ISABus/I8042Controller.cpp

```diff
@@ -210,15 +241,60 @@ UNMAP_AFTER_INIT ErrorOr<void> I8042Controller::detect_devices()
         }
     }
 
+    m_irq_handlers[0] = TRY(I8042ControllerIRQHandler::try_create(*this, IRQ_FIRST_PORT));
+    m_irq_handlers[1] = TRY(I8042ControllerIRQHandler::try_create(*this, IRQ_SECOND_PORT));
+
     // Enable IRQs after both are detected and initialized
     if (m_first_ps2_port.device)
-        m_first_ps2_port.device->enable_interrupts();
+        m_irq_handlers[0]->enable_irq();
     if (m_second_ps2_port.device)
-        m_second_ps2_port.device->enable_interrupts();
+        m_irq_handlers[1]->enable_irq();
     return {};
 }
 
-bool I8042Controller::irq_process_input_buffer(PS2Device::Type instrument_type)
+ErrorOr<void> I8042Controller::send_command(PortIndex port_index, DeviceCommand command)
+{
+    switch (command) {
+    case DeviceCommand::GetDeviceID:
+        return send_command_to_specific_port(port_index, I8042Command::GetDeviceID);
+    case DeviceCommand::EnablePacketStreaming:
+        return send_command_to_specific_port(port_index, I8042Command::EnablePacketStreaming);
+    case DeviceCommand::DisablePacketStreaming:
+        return send_command_to_specific_port(port_index, I8042Command::DisablePacketStreaming);
+    case DeviceCommand::SetDefaults:
+        return send_command_to_specific_port(port_index, I8042Command::SetDefaults);
+
+    // NOTE: The sample rate command is supported only with sending data byte with it!
+    case DeviceCommand::SetSampleRate:
+        return EOPNOTSUPP;
+
+    default:
+        return EINVAL;
+    }
+}
+
+ErrorOr<void> I8042Controller::send_command(PortIndex port_index, DeviceCommand command, u8 data)
+{
+    switch (command) {
+    // NOTE: Only the sample rate command supports sending data byte with it!
+    case DeviceCommand::SetSampleRate:
+        return send_command_to_specific_port(port_index, I8042Command::SetSampleRate, data);
+
+    case DeviceCommand::GetDeviceID:
+        return EOPNOTSUPP;
+    case DeviceCommand::EnablePacketStreaming:
+        return EOPNOTSUPP;
+    case DeviceCommand::DisablePacketStreaming:
+        return EOPNOTSUPP;
+    case DeviceCommand::SetDefaults:
+        return EOPNOTSUPP;
+
+    default:
+        return EINVAL;
+    }
+}
```

Comment:
Why do both these and `send_command_to_specific_port` exist? It looks like `send_command` just forwards to that; there is no extra logic here. Perhaps the flow could be simplified by moving the checks into `do_send_command`.